File: utils/documentCache.ts
// Simple in-memory document cache
interface DocumentCacheEntry {
  text: string;
  fileName: string;
  timestamp: number;
}

class DocumentCache {
  private cache: Map<string, DocumentCacheEntry>;

  constructor() {
    this.cache = new Map();
  }

  store(docId: string, text: string, fileName: string): void {
    this.cache.set(docId, {
      text,
      fileName,
      timestamp: Date.now()
    });
  }

  get(docId: string): DocumentCacheEntry | undefined {
    return this.cache.get(docId);
  }

  exists(docId: string): boolean {
    return this.cache.has(docId);
  }

  clear(): void {
    this.cache.clear();
  }
}

// Export singleton instance
export const documentCache = new DocumentCache(); 

File: utils/document_cache.py
from typing import Dict, Optional
from datetime import datetime

class DocumentCache:
    def __init__(self):
        self._cache: Dict[str, Dict] = {}

    def store(self, doc_id: str, text: str, metadata: Optional[Dict] = None) -> None:
        """Store document in cache"""
        self._cache[doc_id] = {
            "text": text,
            "metadata": metadata or {},
            "timestamp": datetime.now().isoformat()
        }

    def get(self, doc_id: str) -> Optional[Dict]:
        """Get document from cache"""
        return self._cache.get(doc_id)

    def exists(self, doc_id: str) -> bool:
        """Check if document exists in cache"""
        return doc_id in self._cache

    def keys(self) -> list:
        """Get all document IDs in cache"""
        return list(self._cache.keys())

# Global instance
document_cache = DocumentCache() 